jsonrpc/src/common/connection.ts

```diff
@@ -615,21 +626,28 @@ export function createMessageConnection(messageReader: MessageReader, messageWri
 			processMessageQueue();
 		});
 	}
-
+	function handleMessage (message: Message) {
+		if (Message.isRequest(message)) {
+			handleRequest(message);
+		} else if (Message.isNotification(message)) {
+			handleNotification(message);
+		} else if (Message.isResponse(message)) {
+			handleResponse(message);
+		} else {
+			handleInvalidMessage(message);
+		}
+	}
 	function processMessageQueue(): void {
 		if (messageQueue.size === 0) {
 			return;
 		}
 		const message = messageQueue.shift()!;
 		try {
-			if (Message.isRequest(message)) {
-				handleRequest(message);
-			} else if (Message.isNotification(message)) {
-				handleNotification(message);
-			} else if (Message.isResponse(message)) {
-				handleResponse(message);
+			const messageStrategy = options?.messageStrategy;
+			if (messageStrategy && MessageStrategy.is(messageStrategy)) {
```

Comment:
IMo testing that `messageStrategy` is set is not necessary since `MessageStrategy.is(messageStrategy)` will fail if it is so.

Reply:
Good point, updated